### PR TITLE
 Spiral Matrix Program

### LINKED ISSUE #1919: CP Displaying Matrix in Spiral Form
### Description

Documentation on how to display a  matrix in spiral form 

### Domain

Competitive Programming

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### LINKED ISSUE #1921: C/CPP Reversing a word in a string
### Description

Documentation on how reverse a word  in a string

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by yellowberard**
<hr>

## Description 📜

A program to traverse spirally in a matrix.
Fixes #1919

<hr>

## Type of change 📝

- [x] Doc#umentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] Competitive #Programming

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [xI have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

### CODE REVIEW

**Comment by yellowberard** [comment]
@ravireddy07 This is my new PR.

File: Competitive_Programming/C++/Matrix/Spiral_Matrix.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ 
+  matrix[][] = {{1, 2, 3, 4}, {5,6, 7, 8}, {9, 10, 11, 12}, {13, 14, 15,16}} 
+  
+  *Output*: 1 2 3 4 8 12 16 15 14 13 9 5 6 7 11 10
+***
+*Input*: r = 3, c = 4 
+
+ matrix[][] = {{1, 2, 3, 4}, {5, 6, 7, 8}, {9,10, 11, 12}}
+ 
+ *Output*: 1 2 3 4 8 12 11 10 9 5 6 7
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Matrix/Spiral_Matrix.md:24]: To make any Text/word bolder, you have to use 2*.
Above Input/Output words are not formatted to bold.

File: Competitive_Programming/C++/Matrix/Spiral_Matrix.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+left-> right then 
+top-> bottom then 
+left->right then  
+bottom-> top and 
+so on  till we reach the center element.
+I have given a Example for you to get a better understanding:
+
+Given a matrix of size r\*c. Traverse the matrix in spiral form.
+
+## Example:
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Matrix/Spiral_Matrix.md:12]: Change to Input/Output

File: Competitive_Programming/C++/Matrix/Spiral_Matrix.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+ *Input*: r = 4, c = 4
+ 
+  matrix[][] = {{1, 2, 3, 4}, {5,6, 7, 8}, {9, 10, 11, 12}, {13, 14, 15,16}}
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Matrix/Spiral_Matrix.md:16]:
```suggestion
  matrix[][] = `{{1, 2, 3, 4}, {5,6, 7, 8}, {9, 10, 11, 12}, {13, 14, 15,16}} `
```

File: Competitive_Programming/C++/Matrix/Spiral_Matrix.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  *Output*: 1 2 3 4 8 12 16 15 14 13 9 5 6 7 11 10
+***
+*Input*: r = 3, c = 4 
+
+ matrix[][] = {{1, 2, 3, 4}, {5, 6, 7, 8}, {9,10, 11, 12}}
+ 
+ *Output*: 1 2 3 4 8 12 11 10 9 5 6 7
+
+
+## Code
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Matrix/Spiral_Matrix.md:27]: Seems like code is not-formatted or not properly indented. Use 4-Space Indentation and place the code here.
And closing bracket is missing for main method.

File: Competitive_Programming/C++/Matrix/Spiral_Matrix.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+We can explain time complexity by as the loop will keep till total elements of a matrix is covered so total number of elements in a matrix is column*row hence r*c.
+
+**Time Complexity: O(r*c)**
+
+**Space Complexity: O(r*c)**
+## Acknowledgements
+
+ - [Question link](https://practice.geeksforgeeks.org/problems/spirally-traversing-a-matrix-1587115621/1#)
+ - Notes I made while studying
+
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Matrix/Spiral_Matrix.md:114]: No Referenences are allowed. Please remove it.

File: Competitive_Programming/C++/Matrix/Spiral_Matrix.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+## Time Complexity
+We can explain time complexity by as the loop will keep till total elements of a matrix is covered so total number of elements in a matrix is column*row hence r*c.
+
+**Time Complexity: O(r*c)**
+
+**Space Complexity: O(r*c)**
+## Acknowledgements
+
+ - [Question link](https://practice.geeksforgeeks.org/problems/spirally-traversing-a-matrix-1587115621/1#)
+ - Notes I made while studying
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Matrix/Spiral_Matrix.md:113]: If you have any more supporting material which you own the copyright, then just discuss it here with images or place it here by mentioning what it is, what it contains...etc

File: Competitive_Programming/C++/Matrix/Spiral_Matrix.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
+# Displaying a Matrix in Spiral Form 
+ In this question we will display a matrix in spiral form by traversing through 
+left-> right then 
+top-> bottom then 
+left->right then  
+bottom-> top and 
+so on  till we reach the center element.
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Matrix/Spiral_Matrix.md:7]:
```suggestion
 In this question we will display a matrix in spiral form by traversing through
 
 * left-> right then
 * top-> bottom then
 * left->right then
 * bottom-> top and
 
 so on  till we reach the center element.
```

**Review by ravireddy07 — CHANGES_REQUESTED** [review]

File: Competitive_Programming/C++/Matrix/Spiral_Matrix.md (hunk 2 above)
Comment (ravireddy07) [reply Competitive_Programming/C++/Matrix/Spiral_Matrix.md:12]: Place this input/output below the code

**Review by ravireddy07 — CHANGES_REQUESTED** [review]
Use 4-space indentation and place the code and seems like few requested changes haven't changed. Please have a look

File: Competitive_Programming/C++/Matrix/Spiral_Matrix.md (hunk 4 above)
Comment (yellowberard) [reply Competitive_Programming/C++/Matrix/Spiral_Matrix.md:27]: I have tried online formatter but getting the same result.

**Comment by ravireddy07** [comment]
i think you missed the explanation @yellowberard

**Comment by yellowberard** [comment]
I have done indentation also and made all the changes.

**Comment by ravireddy07** [comment]
> I have done indentation also and made all the changes.

I don't see an explanation.

**Comment by yellowberard** [comment]
@ravireddy07 I have added the explanation. Sorry for the delay.

**Comment by ravireddy07** [comment]
just add the formatted code to the doc, The code structure is bad to see @yellowberard

**Comment by yellowberard** [comment]
I have done the indentation in doc and also formatted the code online.

**Review by ravireddy07 — APPROVED** [review]

**Comment by yellowberard** [comment]
@ravireddy07 please accept the  PR and if any changes required please let me know.

**Comment by ravireddy07** [comment]
> @ravireddy07 please accept the PR and if any changes required please let me know.

Update the Status in PR tracksheet, it will be merged.

**Comment by yellowberard** [comment]
@ravireddy07 Where can I find PR track sheet?

**Comment by ravireddy07** [comment]
> @ravireddy07 Where can I find PR track sheet?

In Discord, Check #general channel's pinned messages. or just mention me in that channel...i'll share the link

**Comment by yellowberard** [comment]
@ravireddy07 I have added my PR in The Track sheet.

**Comment by ravireddy07** [comment]
> @ravireddy07 I have added my PR in The Track sheet.
Yup, it will be merged by supervisor. Mention anyone in discord, for faster action.

**Pull request merged**